src/holder.ts

```diff
@@ -1,61 +1,64 @@
-import { JWTHeaderParameters, JWTPayload, KeyLike, SignJWT, decodeJwt, importJWK, jwtVerify } from 'jose';
-import { CreateSDJWTPayload, JWT, PresentSDJWTPayload, SD_JWT_FORMAT_SEPARATOR } from './types';
-import { generateNonce, isValidUrl, supportedAlgorithm } from './util';
+import { KeyBindingVerifier, decodeJWT } from 'sd-jwt';
+import { CreateSDJWTPayload, JWT, PresentSDJWTPayload, SD_JWT_FORMAT_SEPARATOR, SignerConfig } from './types';
+import { generateNonce, isValidUrl } from './util';
 
 export class Holder {
-  private privateKey: KeyLike | Uint8Array;
-  private algorithm: supportedAlgorithm;
+  private signer: SignerConfig;
   private static SD_KEY_BINDING_JWT_TYP = 'kb+jwt';
 
-  constructor(privateKey: KeyLike | Uint8Array, algorithm: supportedAlgorithm) {
-    if (!privateKey) {
-      throw new Error('Holder private key is required');
+  constructor(signer: SignerConfig) {
+    if (!signer?.callback || typeof signer?.callback !== 'function') {
+      throw new Error('Signer function is required');
     }
-    if (!algorithm || typeof algorithm !== 'string') {
-      throw new Error(`Issuer algorithm is required and must be one of ${supportedAlgorithm}`);
+    if (!signer?.algo || typeof signer?.algo !== 'string') {
+      throw new Error('algo used for Signer function is required');
     }
 
-    this.algorithm = algorithm;
-    this.privateKey = privateKey;
+    this.signer = signer;
   }
 
   /**
-   * Gets the key binding JWT for the specified verifier.
+   * Gets a key binding JWT.
    * @param forVerifier The verifier to get the key binding JWT for.
-   * @returns The key binding JWT and nonce.
-   * @throws An error if the key binding JWT could not be created.
+   * @param nonce The nonce to use.
+   * @throws An error if the key binding JWT cannot be created.
+   * @returns The key binding JWT.
    */
-  async getKeyBindingJWT(forVerifier: string): Promise<{ keyBindingJWT: JWT; nonce: string }> {
+  async getKeyBindingJWT(
+    forVerifier: string,
```

Comment:
@vijayshiyani what is this `forVerifier`? Is a strng that will be put as `aud` claim?

Reply:
@linasi  yes, renamed to `aud` with comments, it makes more sense.